network/p2p/p2pnode/libp2pNode.go

```diff
@@ -346,8 +358,29 @@ func (n *Node) WithDefaultUnicastProtocol(defaultHandler libp2pnet.StreamHandler
 // WithPeersProvider sets the PeersProvider for the peer manager.
 // If a peer manager factory is set, this method will set the peer manager's PeersProvider.
 func (n *Node) WithPeersProvider(peersProvider p2p.PeersProvider) {
+	// TODO: chore: we should not allow overriding the peers provider if one is already set.
 	if n.peerManager != nil {
-		n.peerManager.SetPeersProvider(peersProvider)
+		n.peerManager.SetPeersProvider(
+			func() peer.IDSlice {
+				authorizedPeersIds := peersProvider()
+				allowListedPeerIds := peer.IDSlice{} // subset of authorizedPeersIds that are not disallowed
+				for _, peerId := range authorizedPeersIds {
+					// exclude the disallowed peers from the authorized peers list
+					causes, disallowListed := n.disallowListedCache.IsDisallowListed(peerId)
+					if disallowListed {
+						n.logger.Warn().
```

Comment:
With the modifications in [this commit](https://github.com/onflow/flow-go/pull/4441/commits/430b369b25b5fe079822db827a69b68fb56bc666), a single notification is now emitted when a node is disallow-listed, instead of one per heartbeat. The same applies when a node is allow-listed. Consequently, this logging will not be repetitive.